### PR TITLE
apply code static analysis check with phpstan

### PR DESCRIPTION
**Pull request by samsonasik**
apply code static analysis check with phpstan. start with level 0.

**Checklist:**
- [x] Securely signed commits

### CODE REVIEW

**Comment by MGatner** [comment]
My comments from the `phpcs` Actions workflow should apply here as well. 

I know nothing about phpstan so I won't review that portion yet.

**Comment by samsonasik** [comment]
Found a bug when running phpstan in `system/Database/MySQLi/Result.php`: 

```
 ------ ------------------------------------------------------------------------------------- 
  97     Array has 2 duplicate keys with value 1 (MYSQLI_TYPE_TINY, MYSQLI_TYPE_CHAR).        
  111    Array has 2 duplicate keys with value 247 (MYSQLI_TYPE_INTERVAL, MYSQLI_TYPE_ENUM).  
 ------ ------------------------------------------------------------------------------------- 
```

As same index key, it may cause a wrong name on getting data type name.

**Comment by samsonasik** [comment]
@MGatner phpstan run green for level 0 🎉 , ready for review ;)

File: system/API/ResponseTrait.php [hunk 1]
```diff
@@ -102,6 +102,13 @@ trait ResponseTrait
 	 */
 	protected $format = 'json';
 
+	/**
+	 * Formatter instance
+	 *
+	 * @var null|\CodeIgniter\Format\FormatterInterface
+	 */
+	protected $formatter = null;
```
Comment (samsonasik) [inline system/API/ResponseTrait.php:110]: this may be a false positive ref https://3v4l.org/pv1c7

Comment (samsonasik) [reply system/API/ResponseTrait.php]: reverted as false positive for this trait undefined property.

**Pull request merged**